service/src/main.rs

```diff
@@ -329,30 +357,35 @@ impl InclusionService {
                             }
                             Err(e) => {
                                 error!("{job:?} failed progressing DataAvalibile: {e}");
-                                job_status = JobStatus::Failed(e, None);
+                                job_status = JobStatus::Failed(
+                                    e,
+                                    Some(JobStatus::DataAvalibile(proof_input)),
+                                );
                                 self.finalize_job(&job_key, job_status)?;
                             }
                         };
+                        debug!("ZK request sent");
                     }
-                    JobStatus::ZkProofPending(zk_job_id) => {
-                        match self.wait_for_zk_proof(zk_job_id).await {
+                    JobStatus::ZkProofPending(zk_request_id) => {
+                        debug!("ZK request waiting");
+                        match self.wait_for_zk_proof(&job_key, zk_request_id).await {
                             Ok(zk_proof) => {
                                 info!("🎉 {job:?} Finished!");
                                 job_status = JobStatus::ZkProofFinished(zk_proof);
                                 self.finalize_job(&job_key, job_status)?;
                             }
                             Err(e) => {
                                 error!("{job:?} failed progressing ZkProofPending: {e}");
-                                job_status = JobStatus::Failed(e, None);
+                                job_status = JobStatus::Failed(
+                                    e,
+                                    Some(JobStatus::ZkProofPending(zk_request_id)),
+                                );
                                 self.finalize_job(&job_key, job_status)?;
                             }
```

Comment:
While not ideal, as I can't figure out just yet how to handle `anyhow!` errors returned by SP1 SDK in a more nuanced way, we allow retry for anything. Will followup with failure nuance.